cmd/kubeadm/app/cmd/join.go

```diff
@@ -406,12 +404,12 @@ func newJoinData(cmd *cobra.Command, options *joinOptions, out io.Writer) (joinD
 }
 
 // Cfg returns the JoinConfiguration.
-func (j *joinData) Cfg() *kubeadmapi.JoinConfiguration {
+func (j joinData) Cfg() *kubeadmapi.JoinConfiguration {
```

Comment:
Is there any particular reason for changing this? It may be just me, missing something, though...

Reply:
yes there is a reason for it. i explained it on the description of PR. 

look at the point 1 and 2.
But for this, basically the problem is when we need to type-casting the pointer to interface on `phases`

Reply:
My bad for missing this out. Sorry about that.